### PR TITLE
Add createCheckAPI

### PR DESCRIPTION
**Pull request by prettymuchbryce**
As discussed in #527, adds a new public method `createCheckAPI` which allows a user to initialize a new instance of `check`, `checkSchema`, etc using `customValidators` and `customSanitizers`. This greatly improves the usability of the check API. It also happens to solve #552. The goal of this change is to be completely backwards compatible.

TODO
- Additional unit tests
- Typescript definitions
- Documentation

@gustavohenke Would love to get your input on the direction so far.

### CODE REVIEW

File: check/create-check-api.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  customValidators,
+  customSanitizers
+} = {}) => {
+  const customCheck = (fields, locations, message) =>
+    check(
+      fields,
+      locations,
+      message,
+      Object.assign({}, customValidators, validator),
+      Object.assign({}, customSanitizers, validator)
```
Comment (gustavohenke) [inline check/create-check-api.js]: By doing this, one could have their custom validators/sanitizers overwritten by validator's ones.
Could be an astonishment point for users that deliberately wanted to overwrite the standard ones, or for those that didn't know one with the same name exists and so their function never runs.

File: check/validation-chain-builders.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-const check = require('./check');
+const checkModule = require('./check');
 
-const buildCheckFunction = locations => (fields, message) => check(fields, locations, message);
+const buildCheckFunction = locations => (fields, message, check = checkModule) => check(fields, locations, message);
```
Comment (gustavohenke) [inline check/validation-chain-builders.js]: Did you mean to pass `check` to the outer function? `(locations, check = checkModule) => ...`?

File: check/check.js [hunk 3]
```diff
@@ -17,10 +17,10 @@ module.exports = (fields, locations, message) => {
     }, next);
   };
 
-  Object.keys(validator)
+  Object.keys(validatorSet)
```
Comment (gustavohenke) [inline check/check.js]: When `createCheckAPI()` mixes standard validators with custom ones, the custom ones won't be built with the flag `custom: true`.
Internally, this is used by to determine how to call the validator (or sanitizer).

A better option is to always add the standard ones from validator.js, and change `validatorSet`  to simply `customVaildators`, then extend them like it's done a few lines below with `isArray`/`isString`/`exists`.

File: check/validation-chain-builders.js (hunk 2 above)
Comment (prettymuchbryce) [reply check/validation-chain-builders.js]: I'm not sure I understand the question. We need to use our custom version of check in the case that the user uses `createCheckAPI` so it is passed in as an argument. For the normal case, we just use the default argument checkModule for the normal non-custom check module.

If you look at `create-check-api.js` this is where we pass in our custom check function to `buildCheckFunction`.

**Comment by prettymuchbryce** [comment]
@gustavohenke Any recommendations on how to gracefully test this? I have a couple of tests, but ideally we run all of the same tests that touch the exports of `index` with those from `createCheckAPI` instead. I could copy and paste all of the relevant tests, but it seems like a little too much code duplication.

**Review by fedeci — COMMENTED** [review]
Hi @prettymuchbryce, this is definitely a smart solution, but should be reworked to fit the updated codebase😬
Can you handle this?

**Pull request closed without merging**